test/java/seedu/addressbook/commands/DeleteMCommandTest.java

```diff
@@ -0,0 +1,158 @@
+                new Email("sam@doe.com", false), new Address("55G Abc Road", false), Collections.emptySet());
+        Person davidGrant = new Person(new Name("David Grant"), new Phone("61121122", false),
+                new Email("david@grant.com", false), new Address("44H Define Road", false),
+                Collections.emptySet());
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, janeDoe, davidGrant, samDoe);
+
+        emptyDisplayList = TestUtil.createList();
+
+        listWithEveryone = TestUtil.createList(johnDoe, janeDoe, davidGrant, samDoe);
+        listWithSurnameDoe = TestUtil.createList(johnDoe, janeDoe, samDoe);
+    }
+
+    @Test
+    public void execute_emptyAddressBook_returnsPersonNotFoundMessage() {
+        assertDeletionFailsDueToNoSuchPerson(Arrays.asList(1) , emptyAddressBook, listWithEveryone);
+    }
+
+    @Test
+    public void execute_noPersonDisplayed_returnsInvalidIndexMessage() {
+        assertDeletionFailsDueToInvalidIndex(Arrays.asList(1), addressBook, emptyDisplayList);
+    }
+
+    @Test
+    public void execute_targetPersonNotInAddressBook_returnsPersonNotFoundMessage()
+            throws IllegalValueException {
+        Person notInAddressBookPerson = new Person(new Name("Not In Book"), new Phone("63331444", false),
+                new Email("notin@book.com", false), new Address("156D Grant Road", false), Collections.emptySet());
+        List<ReadOnlyPerson> listWithPersonNotInAddressBook = TestUtil.createList(notInAddressBookPerson);
+
+        assertDeletionFailsDueToNoSuchPerson(Arrays.asList(1), addressBook, listWithPersonNotInAddressBook);
+    }
+
+    @Test
+    public void execute_invalidIndex_returnsInvalidIndexMessage() {
+        assertDeletionFailsDueToInvalidIndex(Arrays.asList(0), addressBook, listWithEveryone);
+        assertDeletionFailsDueToInvalidIndex(Arrays.asList(-1), addressBook, listWithEveryone);
+        assertDeletionFailsDueToInvalidIndex(Arrays.asList(listWithEveryone.size() + 1), addressBook, listWithEveryone);
+    }
+
+    @Test
+    public void execute_validIndex_personIsDeleted() throws PersonNotFoundException {
+        assertDeletionSuccessful(Arrays.asList(1), addressBook, listWithSurnameDoe);
+        assertDeletionSuccessful(Arrays.asList(listWithSurnameDoe.size()), addressBook, listWithSurnameDoe);
+
+        int middleIndex = (listWithSurnameDoe.size() / 2) + 1;
+        assertDeletionSuccessful(Arrays.asList(middleIndex), addressBook, listWithSurnameDoe);
+    }
+
+    //TODO: tests can be made more general
+    /**
+     * Creates a new delete command.
+     *
+     * @param targetIndexes of the people that we want to delete
+     */
+    private DeleteMCommand createDeleteMCommand(List<Integer> targetIndexes, AddressBook addressBook,
+                                                                      List<ReadOnlyPerson> displayList) {
```

Comment:
There seems to be a few formatting issues, i.e. lines are truncated

Reply:
Noted. Thank you! I will update in the level3/4.